docs/source/undo.rst

```diff
@@ -0,0 +1,275 @@
+    ``end_macro()`` call).
+
+``end_macro()``
+    This method ends the current macro.
+
+
+ICommand
+........
+
+The ``ICommand`` interface defines the interface that must be implemented by
+any undoable/redoable command.
+
+``data``
+    This optional trait is a reference to the data object that the command
+    operates on.  It is not used by the framework itself.
+
+``name``
+    This Str trait is the name of the command as it will appear in any GUI
+    element (e.g. in the text of an undo and redo menu entry).  It may include
+    ``&`` to indicate a keyboard shortcut which will be automatically removed
+    whenever it is inappropriate.
+
+``__init__(*args)``
+    If the command takes arguments then the command must ensure that deep
+    copies should be made if appropriate.
+
+``do()``
+    This method is called by a command stack to execute the command and to
+    return any result.  The command must save any state necessary for the
+    ``undo()`` and ``redo()`` methods to work.  It is guaranteed that this
+    will only ever be called once and that it will be called before any call
+    to ``undo()`` or ``redo()``.
+
+``undo()``
+    This method is called by a command stack to undo the command.
+
+``redo()``
+    This method is called by a command stack to redo the command and to return
+    any result.
+
+``merge(other)``
+    This method is called by the command stack to try and merge the ``other``
+    command with this one.  True should be returned if the commands were
+    merged.  If the commands are merged then ``other`` will not be placed on
+    the command stack.  A subsequent undo or redo of this modified command
+    must have the same effect as the two original commands.
+
+
+AbstractCommand
+...............
+
+``AbstractCommand`` is an abstract base class that implements the ``ICommand``
+interface.  It provides a default implementation of the ``merge()`` method.
+
+
+CommandAction
+.............
+
+The ``CommandAction`` class is a sub-class of the PyFace ``Action`` class that
+is used to wrap commands.
+
+``command``
+    This callable trait must be set to a factory that will return an object
+    that implements ``ICommand``.  It will be called when the action is invoked
+    and the object created pushed onto the command stack.
+
+``command_stack``
+    This instance trait must be set to the command stack that commands invoked
+    by the action are pushed to.
+
+``data``
+    This optional trait is a reference to the data object that will be passed
+    to the ``command`` factory when it is called.
+
+
+UndoAction
+..........
+
+The ``UndoAction`` class is a canned PyFace action that undoes the last
+command of the active command stack.
+
+
+RedoAction
+..........
+
+The ``RedoAction`` class is a canned PyFace action that redoes the last
+command undone of the active command stack.
+
+
+.. _API: api/index.html
+.. _example: https://svn.enthought.com/enthought/browser/AppTools/trunk/examples/undo/
```

Comment:
This link is clearly out of place here, but I believe it was actually previously broken on apptools as well.  Likewise, this final section of "API Overview" seems unnecessary with auto built api docs.  This file needs to be updated, but I think I will leave those changes for a separate PR unless a reviewer would like to see them done here, in which case I am happy to do so

Reply:
Let's get them fixed in this PR instead of after the PR.